### PR TITLE
Authhelper: browser based authentication

### PR DESCRIPTION
**Pull request by psiinon**

### CODE REVIEW

**Comment by psiinon** [comment]
Pending more testing, unit tests and help 😉

**Comment by psiinon** [comment]
This attempts to spot auth tokens by matching known JSON keys.
But I've also now changed it to spot Auth req headers and then loop back through the reqs made as part of auth to try to spot where they came from.
Also added stats so we can spot commonly used keys 😁 
No more changes planned just yet..

**Review by thc202 — APPROVED** [review]

File: addOns/authhelper/src/main/resources/org/zaproxy/addon/authhelper/resources/Messages.properties [hunk 1]
```diff
@@ -7,6 +7,13 @@ The 'Other Info' field contains set of key=value lines which identify any releva
 If the request is in a context which is set to "Manual Authentication" then this rule will attempt to change the authentication to match the request identified.
 authhelper.auth-detect.soln = This is an informational alert rather than a vulnerability and so there is nothing to fix.
 
+authhelper.auth.method.browser.name = Browser-based Authentication
+authhelper.auth.method.browser.label.browser = Browser:
+authhelper.auth.method.browser.label.loginUrl = Login URL *:
+authhelper.auth.method.browser.label.loginWait = Login Wait in Seconds:
+
+authhelper.auth.method.browser.output.sessionid = Session token identified in href {0}
```
Comment (kingthorin) [inline addOns/authhelper/src/main/resources/org/zaproxy/addon/authhelper/resources/Messages.properties:15]: HREF in full caps?

**Comment by psiinon** [comment]
Thanks - hopefully fixed all of those..

**Comment by psiinon** [comment]
Changed to WIP - just found out ZAP is no longer showing the response in the relevant tab :P

**Comment by psiinon** [comment]
Now fixed 😉

**Review by kingthorin — APPROVED** [review]

**Pull request merged**